plugins/woo-ai/src/product-name/product-name-suggestions.tsx

```diff
@@ -177,42 +206,61 @@ export const ProductNameSuggestions = () => {
 		}
 	};
 
+	const buildPrompt = () => {
+		const validProductData = Object.entries( {
+			name: getProductName(),
+			categories: getCategories(),
+			tags: getTags(),
+			attributes: getAttributes(),
+			product_type: getProductType(),
+			is_downloadable: isProductDownloadable(),
+			is_virtual: isProductVirtual(),
+		} ).reduce( ( acc, [ key, value ] ) => {
+			if (
+				typeof value === 'boolean' ||
+				( value instanceof Array
+					? Boolean( value.length )
+					: Boolean( value ) )
+			) {
+				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+				// @ts-ignore
+				acc[ key ] = value;
+			}
+			return acc;
+		}, {} );
+
+		const instructions = [
+			'You are a WooCommerce SEO and marketing expert.',
+			"Using the product's name, description, tags, categories, and other attributes, provide three optimized alternatives to the product's title to enhance the store's SEO performance and sales.",
+			"Provide the best option for the product's title based on the product properties.",
+			'Identify the language used in the given title and use the same language in your response.',
+			'Return only the alternative value for product\'s title in the "content" part of your response.',
+			'Product titles should contain at least 20 characters.',
+			'Return a short and concise reason for each suggestion in seven words in the "reason" part of your response.',
+			"The product's properties are:",
+			`${ JSON.stringify( validProductData ) }`,
```

Comment:
Instead of passing JSON in the prompt, could we pass a comma-separated "key: value" list? It would also be helpful to convert booleans to Yes/No values and remove any underlines from the keys. It might save us a few tokens and be more similar to natural language.

Reply:
Right, we could remove the brackets, although it would add in several newline characters, so I'm not sure it would save us any tokens. I had the thought that ChatGPT may actually like this to be more "structured," and it cleanly delineates it from the rest of the prompt. That said, I'm a little ambivalent, although it would make the code a bit more complex 🤔